.github/workflows/autotag.yaml

```diff
@@ -0,0 +1,15 @@
+name: Bump version
+on:
+  push:
+    branches:
+      - master
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Bump version and push tag
+        id: tag_version
+        uses: mathieudutour/github-tag-action@v6.2
```

Comment:
I'd give https://github.com/googleapis/release-please a try which we use for Helm Chart release in Cosmo. It maintains release PR's for changes and creates a release summary and github release based on conventional commits. It also takes care of tagging.